File: day-02/part-2/julesdt.go
package main

import (
	"fmt"
	"io/ioutil"
	"os"
	"strconv"
	"strings"
	"time"
)

func run(s string) interface{} {
	total := 0
	for _, line := range strings.Split(s, "\n") {
		policySplit := strings.Split(line, ": ")
		password := policySplit[1]
		numbersSplit := strings.Split(policySplit[0], " ")
		letter := numbersSplit[1]
		numbers := strings.Split(numbersSplit[0], "-")
		low, _ := strconv.Atoi(numbers[0])
		high, _ := strconv.Atoi(numbers[1])
		if (string(password[low-1]) == letter) != (string(password[high-1]) == letter) {
			total++
		}
	}
	return total
}

func main() {
	// Uncomment this line to disable garbage collection
	// debug.SetGCPercent(-1)

	// Read input from stdin
	input, err := ioutil.ReadAll(os.Stdin)
	if err != nil {
		panic(err)
	}

	// Start resolution
	start := time.Now()
	result := run(string(input))

	// Print result
	fmt.Printf("_duration:%f\n", time.Now().Sub(start).Seconds()*1000)
	fmt.Println(result)
}

File: day-02/part-2/julesdt.py
from tool.runners.python import SubmissionPy


class JulesdtSubmission(SubmissionPy):

    def run(self, s):
        """
        :param s: input in string format
        :return: solution flag
        """
        password_valid = 0
        for line in s.split('\n'):
            policy, password = line.split(': ')
            numbers, letter = policy.split(' ')
            low, high = map(int, numbers.split('-'))
            first = password[low-1] == letter
            second = password[high-1] == letter
            if first ^ second:
                password_valid += 1
        return password_valid
